%YAML:1.0
#--------------------------------------------------------------------------------------------
# Camera Parameters
#--------------------------------------------------------------------------------------------

# Flag do undistortion
flagDoUndistortion: 1 # 0: do not undistort an image (KITTI), 1: undistort an image

# Camera calibration and distortion parameters (OpenCV) 
# Left camera
Camera.left.fx: 667.959809653146
Camera.left.fy: 667.713938545685
Camera.left.cx: 545.634424235588
Camera.left.cy: 395.629849906665

Camera.left.k1: -0.321306365737294
Camera.left.k2: 0.168216630982978
Camera.left.k3: 0.0583388111983407
Camera.left.p1: 0.0
Camera.left.p2: 0.0

Camera.left.width: 1032
Camera.left.height: 772

# Right camera
Camera.right.fx: 668.237173199852
Camera.right.fy: 668.199813476523
Camera.right.cx: 522.375134461151
Camera.right.cy: 409.972277738036

Camera.right.k1: -0.319454818441647
Camera.right.k2: 0.160425305072180
Camera.right.k3: -0.0506131487392573
Camera.right.p1: 0.0
Camera.right.p2: 0.0

Camera.right.width: 1032
Camera.right.height: 772

T_lr: !!opencv-matrix # this statement is necessary.
  rows: 4
  cols: 4
  dt: f
  data: [0.999999025791103, -0.00137955561858292, -0.000212704348221640, 0.254740180977306, 0.00137997239354495, 0.999997103690113, 0.00197187412818722, 8.59566308727688e-05, 0.000209983422131205, -0.00197216573329844, 0.999998033232707, -0.000641735851620330, 0, 0, 0, 1]

#--------------------------------------------------------------------------------------------
# User setting Parameters
#--------------------------------------------------------------------------------------------

# Parameters of feature tracker
feature_tracker.thres_error: 40.0      
feature_tracker.thres_bidirection: 0.5
feature_tracker.thres_sampson: 1.0
feature_tracker.window_size: 15
feature_tracker.max_level: 6

# Parameters of map update
map_update.thres_parallax: 1.0

# Parameters of feature extractor
feature_extractor.n_features: 2000
feature_extractor.n_bins_u: 32
feature_extractor.n_bins_v: 20
feature_extractor.thres_fastscore: 10.0
feature_extractor.radius: 5.0

# Parameters of motion estimator
motion_estimator.thres_1p_error: 120.0 # pixels
motion_estimator.thres_5p_error: 1.0 # pixels
motion_estimator.thres_poseba_error: 5.0 # pixels

motion_estimator.lba.max_iter: 6

# Parameters of keyframe update
keyframe_update.thres_alive_ratio: 0.7
keyframe_update.thres_mean_parallax: 1.0
keyframe_update.thres_trans: 1.0 # meters
keyframe_update.thres_rotation: 4.0 # degrees